Type specification seems not to permit defining a schema that contains a list that is of a pyarrow field type nor does it permit types such as pyarrow.timestamp

Example python file

```import pyarrow


thing = pyarrow.schema(
    [
        pyarrow.field("eg", pyarrow.string()),
        pyarrow.field("testcase", pyarrow.list_(pyarrow.string())),
    ]
)
thing_two = pyarrow.schema(
    [
        pyarrow.field("eg", pyarrow.string()),
        pyarrow.field("timestamp", pyarrow.timestamp("s")),
    ]
)
```

^ this should be permitted. However, we run mypy 

```
mypy testyx.py
testyx.py:5: error: Argument 1 to "schema" has incompatible type "list[object]"; expected "Iterable[Field[Any]] | Iterable[tuple[str, DataType]] | Mapping[str, DataType]"  [arg-type]
testyx.py:11: error: Argument 1 to "schema" has incompatible type "list[object]"; expected "Iterable[Field[Any]] | Iterable[tuple[str, DataType]] | Mapping[str, DataType]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```



```
pip list|grep stub
pyarrow-stubs                          20.0.0.20250825

pip list|grep pyarrow
pyarrow                                21.0.0
pyarrow-stubs                          20.0.0.20250825


pip list|grep mypy
mypy                                   1.17.1
mypy_extensions                        1.1.0
```

--- a/pyarrow-stubs/__lib_pxi/types.pyi
+++ b/pyarrow-stubs/__lib_pxi/types.pyi
@@ -1549,7 +1549,9 @@ def ensure_metadata(
     meta: Mapping[bytes | str, bytes | str] | KeyValueMetadata | None, allow_none: bool = False
 ) -> KeyValueMetadata | None: ...
 
-class Field(_Weakrefable, Generic[_DataTypeT]):
+_DataTypeT_co = TypeVar("_DataTypeT_co", bound=DataType, covariant=True)
+
+class Field(_Weakrefable, Generic[_DataTypeT_co]):
     """
     A named field, with a data type, nullability, and optional metadata.
 
@@ -1647,7 +1649,7 @@ class Field(_Weakrefable, Generic[_DataTypeT]):
         {b'key': b'Something important'}
         """
     @property
-    def type(self) -> _DataTypeT: ...
+    def type(self: Field[_DataTypeT]) -> _DataTypeT: ...
     def with_metadata(self, metadata: dict[bytes | str, bytes | str]) -> Self:
         """
         Add metadata as dict of string keys and values to Field
@@ -1744,7 +1746,7 @@ class Field(_Weakrefable, Generic[_DataTypeT]):
         >>> field_new
         pyarrow.Field<lock: int32>
         """
-    def with_nullable(self, nullable: bool) -> Field[_DataTypeT]:
+    def with_nullable(self: Field[_DataTypeT], nullable: bool) -> Field[_DataTypeT]:
         """
         A copy of this field with the replaced nullability
 

